src/main/java/beam/utils/LoggingUtil.java

```diff
@@ -30,24 +34,23 @@ public static void initLogger(String outputDirectory, boolean keepConsoleAppende
             }
             // ################################################
 
-            InputStream resourceAsStream = context.getClass().getClassLoader().getResourceAsStream(logFileName);
+            InputStream resourceAsStream = context.getClass().getClassLoader().getResourceAsStream(logbackConfigFile);
             if (resourceAsStream != null) {
                 try {
+                    context.reset();
```

Comment:
Why did you move the these lines upper? According the documentation http://logback.qos.ch/faq.html#sharedConfiguration
```java
LoggerContext context = (LoggerContext) LoggerFactory.getILoggerFactory();
  JoranConfigurator jc = new JoranConfigurator();
  jc.setContext(context);
  context.reset(); // override default configuration
  // inject the name of the current application as "application-name"
  // property of the LoggerContext
  context.putProperty("application-name", NAME_OF_CURRENT_APPLICATION);
  jc.doConfigure("/path/to/the/above/configuration/file.xml"); 
```

Reply:
just code organization. Does not makes difference since `jc.setContext(context);` only saves the reference
```Java
    public void setContext(Context context) {
        if (this.context == null) {
            this.context = context;
        } else if (this.context != context) {
            throw new IllegalStateException("Context has been already set");
        }
    }
```